packages/realm/src/assert.ts

```diff
@@ -98,10 +98,14 @@ assert.symbol = (value: unknown, name?: string): asserts value is symbol => {
 assert.object = <K extends string | number | symbol = string, V = unknown>(
   value: unknown,
   name?: string,
+  allowArrays = true,
```

Comment:
I would also expect this to account for only objects and not arrays (especially since `assert.array` has also been defined). Although, the original code did allow for arrays and I didn't want to change other parts of the code that wasn't a part of the normalization. So I added `allowArrays` as a parameter to work with the new code.

But I can look into refactoring the other code so that we can either:
* (a) Default `allowArrays` to `false`, or
* (b) Have `assert.object` only account for objects (not null or arrays) and introduce `assert.objectOrArray` that the other code can use.

Reply:
Slight preference for b, but a is also fine with me.

Reply:
b makes the names more consistent with the behavior as well

Reply:
Sorry, I didn't finish reading b closely enough. I don't think we will even need `assert.objectOrArray` if you end up getting rid of the `normalizeObjectSchema` array case.

Reply:
Tests in Milestone 5 fail if `assert.object` does not allow arrays, and the `flags.ALLOW_VALUES_ARRAYS` is `false`. So apparently, `assert.object` is used elsewhere to assert arrays as well. Will try to pinpoint where this happens once I've incorporated the rest of the feedback and see if that can be safely refactored.